### PR TITLE
Update manifests.yml to dynamically define jdk versions based on manifest image jdk export

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Update manifests.yml to dynamically define jdk versions based on manifest image jdk export

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-build/issues/4227

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/manifests.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@
 name: manifests
 
 on:
+  push:
+    branches:
+      - 'manifests-jdks'
```
Comment (rishabh6788) [inline .github/workflows/manifests.yml:7]: Some background regarding this branch would help.

Comment (peterzhuamazon) [reply .github/workflows/manifests.yml]: I forget. removed.

File: .github/workflows/manifests.yml [hunk 2]
```diff
@@ -28,45 +28,26 @@ jobs:
...
-      PYTHON_VERSION: 3.9
-      JDK_VERSION: 17
-    strategy:
-      matrix: ${{ fromJson(needs.list-changed-manifests.outputs.matrix) }}
-    steps:
-      - uses: actions/checkout@v3
-      - name: Set Up JDK ${{ env.JDK_VERSION }}
+          java_version=`cat ${{ matrix.manifest }} | yq -r .ci.image.args | grep -Eo '[0-9]+' || echo ''`
+          echo $java_version
+          echo "JAVA_VERSION=$java_version" >> "$GITHUB_ENV"
```
Comment (rishabh6788) [inline .github/workflows/manifests.yml:45]: use `export` or GITHUB_ENV takes care of that?

Comment (peterzhuamazon) [reply .github/workflows/manifests.yml:45]: It is across steps so need to use $GITHUB_ENV.

File: .github/workflows/manifests.yml [hunk 3]
```diff
@@ -28,45 +28,26 @@ jobs:
...
-    runs-on: ubuntu-latest
-    env:
-      PYTHON_VERSION: 3.9
-      JDK_VERSION: 17
-    strategy:
-      matrix: ${{ fromJson(needs.list-changed-manifests.outputs.matrix) }}
-    steps:
-      - uses: actions/checkout@v3
-      - name: Set Up JDK ${{ env.JDK_VERSION }}
+          java_version=`cat ${{ matrix.manifest }} | yq -r .ci.image.args | grep -Eo '[0-9]+' || echo ''`
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply .github/workflows/manifests.yml:43]: It is. If there is jdk just install, if not, skip, which include osd does not need jdk.

**Comment by peterzhuamazon** [comment]
@rishabh6788 test example:
https://github.com/peterzhuamazon/opensearch-build/actions/runs/7064094940/job/19231436486

**Review by rishabh6788 — APPROVED** [review]

**Pull request merged**